### ISSUE TITLE
model-missing-unicode false positive for subclassed model

### ISSUE THREAD

**Issue by edmorley** [PyCQA/pylint-django#149]
Hi

For this code block ([original source](https://github.com/mozilla/treeherder/blob/572c75d8ecdbbd385f1177284547d8410401140c/treeherder/model/models.py#L40-L42)):

```python
from django.db import models
from django.utils.encoding import python_2_unicode_compatible


@python_2_unicode_compatible
class NamedModel(models.Model):
    id = models.AutoField(primary_key=True)
    name = models.CharField(max_length=100, unique=True)

    class Meta:
        abstract = True

    def __str__(self):
        return self.name


class Product(NamedModel):
    class Meta:
        db_table = 'product'
```

pylint with pylint_django reports:
```
treeherder/model/models.py:40,0: No __unicode__ method on model (Product) (W5101: model-missing-unicode)
```

The parent model already uses the `@python_2_unicode_compatible` decorator, so it seems that instead of W5101, this should be W5103?

The above was using:
Python 2.7.14
Django==1.11.12
pylint==1.8.4
pylint-django==0.9.4
pylint-plugin-utils==0.2.6
astroid==1.6.3

**Comment by atodorov** [comment]
I believe this is fixed with the latest version from master. See #145 .
